File: API/Features/Billing/DocumentTypes/Dtos/DocumentTypeWriteDto.cs
﻿using API.Infrastructure.Interfaces;

namespace API.Features.Billing.DocumentTypes {

    public class DocumentTypeWriteDto : IMetadata {

        // PK
        public int Id { get; set; }
        // Fields
        public string Abbreviation { get; set; }
        public string Description { get; set; }
        public string Batch { get; set; }
        public string LastDate { get; set; }
        public int LastNo { get; set; }
        public bool IsActive { get; set; }
        public string Customers { get; set; }
        public string Suppliers { get; set; }
        // myData
        public bool IsMyData { get; set; }
        public string Table8_1 { get; set; }
        public string Table8_8 { get; set; }
        public string Table8_9 { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/billing/documentTypes/classes/dtos/documentType-write-dto.ts
export interface DocumentTypeWriteDto {

    // PK
    id: number
    // FKs
    shipId: number
    // Fields
    abbreviation: string
    description: string
    batch: string
    lastDate: string
    lastNo: number
    customers: string
    suppliers: string
    discriminatorId: number
    isMyData: boolean
    table8_1: string
    table8_8: string
    table8_9: string
    isActive: boolean
    // Metadata
    putAt: string

}
